hail/python/test/hailtop/test_aiogoogle.py

```diff
@@ -73,8 +77,10 @@ async def test_compose():
 
     async with GoogleStorageClient() as client:
         for i, b in enumerate(part_data):
-            async with await client.insert_object(bucket, f'{token}/{i}') as f:
-                await f.write(b)
+            async def upload(i, b):
+                async with await client.insert_object(bucket, f'{token}/{i}') as f:
+                    await f.write(b)
+            await retry_transient_errors(upload, i, b)
```

Comment:
Bit of a nit but this function either doesn't need inputs or should be defined outside the for loop. My guess is you intended to define this outside the loop.

Reply:
I can't elide the inputs because Python closures don't capture values, they capture references to the variables. I moved outside the for loop.